src/gazebo_magnetometer_plugin.cpp

```diff
@@ -195,18 +207,6 @@ void MagnetometerPlugin::OnUpdate(const common::UpdateInfo&)
     magnetic_field->set_z(magnetic_field_I[2]);
     mag_message_.set_allocated_magnetic_field(magnetic_field);
 
-    for (auto i = 0; i < 9; ++i) {
-      switch (i){
-        // principal diagonal = the variance of the random variables
-        // = noise_density²
-        case 0: case 4: case 8:
-          mag_message_.add_magnetic_field_covariance(noise_density_ * noise_density_);
-          break;
-        default:
-          mag_message_.add_magnetic_field_covariance(0.0);
-      }
-    }
-
```

Comment:
Wouldn't it be better if we should keep it where it is, but use `set_magnetic_field_covariance` instead of `add_magnetic_field_covariance`?

Reply:
@Jaeyoung-Lim actually I agree with this solution because `noise_density_` is loaded once in `Load` and doesn't change after that.